sympy/solvers/ode/nonhomogeneous.py

```diff
@@ -0,0 +1,512 @@
+r"""
+This File contains helper functions for nth_linear_constant_coeff_undetermined_coefficients,
+nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients,
+nth_linear_constant_coeff_variation_of_parameters,
+and nth_linear_euler_eq_nonhomogeneous_variation_of_parameters.
+
+All the functions in this file are used by more than one solvers so, instead of creating
+instances in other classes for using them it is better to keep it here as separate helpers.
+
+Functions that are for internal use:
+
+1) _test_term(coeff, func, order) - It takes coefficient of partiular term, function and
+    order to which it is associated. It returns whether given term matches with linear euler ODEs.
```

Comment:
I don't think that you need to detail each function at the top here. These kinds of comments/description are better placed in or immediately before the function.

This module level docstring just needs to explain the general purpose of the module.